snuba/query/snql/parser.py

```diff
@@ -360,6 +397,14 @@ def visit_entity_name(self, node: Node, visited_children: Tuple[Any]) -> EntityK
         except Exception:
             raise ParsingException(f"{node.text} is not a valid entity name")
 
+    def visit_storage_name(
+        self, node: Node, visited_children: Tuple[Any]
+    ) -> StorageKey:
+        try:
+            return StorageKey(node.text)
+        except Exception:
+            raise ParsingException(f"{node.text} is not a valid Storage name")
```

Comment:
unrelated to current PR, but one thought I have in general is that it might be nice to be able to parse StorageKey and EntityKey even if it doesnt exist. And then later on you get an error if you try to access one that doesnt exist. This would make using fake ones in testing easier.